src/backend/replication/walproposer.c

```diff
@@ -1426,18 +1529,21 @@ AdvancePollState(int i, uint32 events)
 				Assert(wk->voteResponse.term == propTerm);
 
 				/* Handshake completed, do we have quorum? */
-
-				if (++n_votes != quorum)
+				n_votes++;
+				if (n_votes < quorum)
 				{
+					wk->state = SS_IDLE; /* can't do much yet, no quorum */
+				}
+				else if (n_votes > quorum)
+				{
+
+					/* recovery already performed, just start streaming */
+
 					/* Can't start streaming earlier than truncateLsn */
 					wk->startStreamingAt = truncateLsn;
```

Comment:
Not required anymore, we're setting `startStreamingAt` in `SendProposerElected` now.

Reply:
thanks, it actually revealed a bug